crates/data_components/src/delta_lake.rs

```diff
@@ -719,6 +738,278 @@ async fn get_parquet_access_plan(
+        ScalarValue::TimestampSecond(Some(v), None) => Some(Scalar::TimestampNtz(v * 1_000_000)), // Convert to microseconds
+        ScalarValue::TimestampMillisecond(Some(v), Some(_)) => Some(Scalar::Timestamp(v * 1000)), // Convert to microseconds
+        ScalarValue::TimestampMillisecond(Some(v), None) => Some(Scalar::TimestampNtz(v * 1000)), // Convert to microseconds
+        ScalarValue::TimestampMicrosecond(Some(v), Some(_)) => Some(Scalar::Timestamp(v)),
+        ScalarValue::TimestampMicrosecond(Some(v), None) => Some(Scalar::TimestampNtz(v)),
+        ScalarValue::TimestampNanosecond(Some(v), Some(_)) => Some(Scalar::Timestamp(v / 1000)), // Convert to microseconds
+        ScalarValue::TimestampNanosecond(Some(v), None) => Some(Scalar::TimestampNtz(v / 1000)), // Convert to microseconds
+        ScalarValue::TimestampSecond(None, Some(_))
+        | ScalarValue::TimestampMillisecond(None, Some(_))
+        | ScalarValue::TimestampMicrosecond(None, Some(_))
+        | ScalarValue::TimestampNanosecond(None, Some(_)) => Some(Scalar::Null(
+            delta_kernel::schema::DataType::Primitive(PrimitiveType::Timestamp),
+        )),
+        ScalarValue::TimestampSecond(None, None)
+        | ScalarValue::TimestampMillisecond(None, None)
+        | ScalarValue::TimestampMicrosecond(None, None)
+        | ScalarValue::TimestampNanosecond(None, None) => Some(Scalar::Null(
+            delta_kernel::schema::DataType::Primitive(PrimitiveType::TimestampNtz),
+        )),
+        ScalarValue::Null
+        | ScalarValue::Decimal256(_, _, _)
+        | ScalarValue::FixedSizeList(_)
+        | ScalarValue::List(_)
+        | ScalarValue::LargeList(_)
+        | ScalarValue::Struct(_)
+        | ScalarValue::Map(_)
+        | ScalarValue::Time32Second(_)
+        | ScalarValue::Time32Millisecond(_)
+        | ScalarValue::Time64Microsecond(_)
+        | ScalarValue::Time64Nanosecond(_)
+        | ScalarValue::IntervalYearMonth(_)
+        | ScalarValue::IntervalDayTime(_)
+        | ScalarValue::IntervalMonthDayNano(_)
+        | ScalarValue::DurationSecond(_)
+        | ScalarValue::DurationMillisecond(_)
+        | ScalarValue::DurationMicrosecond(_)
+        | ScalarValue::DurationNanosecond(_)
+        | ScalarValue::Union(_, _, _)
+        | ScalarValue::Dictionary(_, _) => None,
+    }
+}
+
+/// Convert a list of `DataFusion` filter expressions to a single `delta_kernel` expression
+///
+/// This function processes multiple `DataFusion` expressions and returns a predicate for `delta_kernel`.
+fn filters_to_delta_kernel_expr(filters: &[Expr]) -> Option<ExpressionRef> {
+    if filters.is_empty() {
+        return None;
+    }
+
+    let mut exprs = Vec::new();
+    for filter in filters {
+        if let Some(expr) = to_delta_kernel_expr(filter) {
+            exprs.push(expr);
+        }
+    }
+
+    if exprs.is_empty() {
+        None
+    } else if exprs.len() == 1 {
+        let expr = exprs.into_iter().next()?;
```

Comment:
```suggestion
        let expr = exprs.pop()?;
```

nit: Not sure why we're turning this into an iter when we can just pop the only item out?